File: packages/js-sdk/src/agents.ts
import { AgenticCRM } from './client'

export type AgentRole = 'lead_sourcing' | 'research' | 'outreach' | 'follow_up' | 'qualification' | 'reporting'
export type AgentStatus = 'active' | 'paused' | 'stopped'

export interface Agent {
  id: number
  name: string
  role: AgentRole
  status: AgentStatus
  config: Record<string, unknown>
  description?: string
  created_at: string
  updated_at: string
}

export interface CreateAgentInput {
  name: string
  role: string
  description?: string
  config?: Record<string, unknown>
}

export interface UpdateAgentInput {
  name?: string
  role?: string
  description?: string
  config?: Record<string, unknown>
}

export class AgentsClient {
  constructor(private client: AgenticCRM) {}

  async list(): Promise<Agent[]> {
    return this.client.get<Agent[]>('/api/agents')
  }

  async get(agentId: number): Promise<Agent> {
    return this.client.get<Agent>(`/api/agents/${agentId}`)
  }

  async create(data: CreateAgentInput): Promise<Agent> {
    return this.client.post<Agent>('/api/agents', data)
  }

  async update(agentId: number, data: UpdateAgentInput): Promise<Agent> {
    return this.client.patch<Agent>(`/api/agents/${agentId}`, data)
  }

  async updateStatus(agentId: number, status: AgentStatus): Promise<Agent> {
    return this.client.patch<Agent>(`/api/agents/${agentId}`, { status })
  }

  async pause(agentId: number): Promise<Agent> {
    return this.updateStatus(agentId, 'paused')
  }

  async resume(agentId: number): Promise<Agent> {
    return this.updateStatus(agentId, 'active')
  }
}

File: packages/python-sdk/src/agentic_crm_sdk/agents.py
"""
Agents API client module.
"""
from __future__ import annotations

from typing import TYPE_CHECKING, Any

if TYPE_CHECKING:
    from agentic_crm_sdk.async_client import AsyncAgenticCRM

from agentic_crm_sdk.client import AgenticCRM


class AgentsClient:
    def __init__(self, client: AgenticCRM):
        self._client = client

    def list(self, **params: Any) -> Any:
        return self._client.get("/api/agents", params=params)

    def get(self, agent_id: int) -> Any:
        return self._client.get(f"/api/agents/{agent_id}")

    def update_status(self, agent_id: int, status: str) -> Any:
        return self._client.patch(f"/api/agents/{agent_id}", json={"status": status})

    def pause(self, agent_id: int) -> Any:
        return self.update_status(agent_id, "paused")

    def resume(self, agent_id: int) -> Any:
        return self.update_status(agent_id, "active")


class AsyncAgentsClient:
    def __init__(self, client: AsyncAgenticCRM):
        self._crm = client

    async def list(self, **params: Any) -> Any:
        return await self._crm.get("/api/agents", params=params)

    async def get(self, agent_id: int) -> Any:
        return await self._crm.get(f"/api/agents/{agent_id}")

    async def update_status(self, agent_id: int, status: str) -> Any:
        return await self._crm.patch(f"/api/agents/{agent_id}", json={"status": status})

    async def pause(self, agent_id: int) -> Any:
        return await self.update_status(agent_id, "paused")

    async def resume(self, agent_id: int) -> Any:
        return await self.update_status(agent_id, "active")